src/util/common_covmat_ops.jl

```diff
@@ -60,7 +60,8 @@ Xt_invA_Y(X::AbstractVecOrMat, A::Cholesky, Y::AbstractVecOrMat) = (A.U' \ X)' *
 
 At_A(A::AbstractVecOrMat) = A'A
 
-diag_At_A(A::AbstractVecOrMat) = vec(sum(abs2.(A); dims=1))
+diag_At_A(A::AbstractMatrix) = vec(sum(abs2.(A); dims=1))
```

Comment:
While we're improving this,
```suggestion
diag_At_A(A::AbstractMatrix) = vec(sum(abs2, A); dims=1))
```
is also about a factor 4-5x faster (40-50µs vs 230-260µs on my laptop for a 500x500 matrix), and avoids allocations.